### PR TITLE
[Docs] New DiT model card

### PR DESCRIPTION
**Pull request by Vixel2006**
# What does this PR do?

This PR updates the model-card for the DiT model, as described in https://github.com/huggingface/transformers/issues/36979, in an attempt to standardize all model-cards.

## Who can review?

[@stevhliu ](https://github.com/stevhliu)

### CODE REVIEW

File: docs/source/en/model_doc/dit.md [hunk 1]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
-- document image classification: the [RVL-CDIP](https://www.cs.cmu.edu/~aharley/rvl-cdip/) dataset (a collection of
-  400,000 images belonging to one of 16 classes).
-- document layout analysis: the [PubLayNet](https://github.com/ibm-aur-nlp/PubLayNet) dataset (a collection of more
-  than 360,000 document images constructed by automatically parsing PubMed XML files).
-- table detection: the [ICDAR 2019 cTDaR](https://github.com/cndplab-founder/ICDAR2019_cTDaR) dataset (a collection of
-  600 training images and 240 testing images).
+# DiT
 
-The abstract from the paper is the following:
+[DiT: Self-supervised Pre-training for Document Image Transformer](https://arxiv.org/abs/2203.02378) by Junlong Li, Yiheng Xu, Tengchao Lv, Lei Cui, Cha Zhang, Furu Wei.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
[DiT](https://huggingface.co/papers/2203.02378) is an image transformer pretrained on large-scale unlabeled document images. It learns to predict the missing visual tokens from a corrupted input image. The pretrained DiT model can now be used as a backbone in other models for visual document tasks like document image classification and table detection.
```

File: docs/source/en/model_doc/dit.md [hunk 2]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
+The DiT model is a Transformer-based architecture designed to process document images (like scanned PDFs, forms, invoices) instead of natural images. The key idea is to pre-train the Transformer in a self-supervised way, so it learns to understand the layout, text, and structure of documents without needing labeled data.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
```

File: docs/source/en/model_doc/dit.md [hunk 3]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/dit_architecture.jpg"
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md:41]: Lets keep the image

File: docs/source/en/model_doc/dit.md [hunk 4]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
-alt="drawing" width="600"/> 
+You can find all the original DiT checkpoints under the [Microsoft](https://huggingface.co/microsoft) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
You can find all the original DiT checkpoints under the [Microsoft](https://huggingface.co/microsoft?search_models=dit) organization.
```

File: docs/source/en/model_doc/dit.md [hunk 5]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
 
-<small> Summary of the approach. Taken from the [original paper](https://arxiv.org/abs/2203.02378). </small>
+> [!TIP]
+> Click on the DiT models in the right sidebar for more examples of how to apply DiT to different document image processing tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
> Refer to the [BEiT](./beit) docs for more examples of how to apply DiT to different vision tasks.
```

File: docs/source/en/model_doc/dit.md [hunk 6]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr). The original code can be found [here](https://github.com/microsoft/unilm/tree/master/dit).
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
The example below demonstrates how to classify an image with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/dit.md [hunk 7]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
-## Usage tips
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-One can directly use the weights of DiT with the AutoModel API:
 
-```python
-from transformers import AutoModel
+```py
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md:42]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="image-classification",
    model="microsoft/dit-base-finetuned-rvlcdip",
    torch_dtype=torch.float16,
    device=0
)
pipeline(images="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/dit-example.jpg")
```

File: docs/source/en/model_doc/dit.md [hunk 8]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
+print(f"Predicted class: {predicted_class}")
 ```
 
-You can also load a fine-tuned model from the [hub](https://huggingface.co/models?other=dit), like so:
+</hfoption>
+<hfoption id="AutoModel">
 
-```python
-from transformers import AutoModelForImageClassification
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md:56]:
```py
import torch
import requests
from PIL import Image
from transformers import AutoModelForImageClassification, AutoImageProcessor

image_processor = AutoImageProcessor.from_pretrained(
    "microsoft/dit-base-finetuned-rvlcdip",
    use_fast=True,
)
model = AutoModelForImageClassification.from_pretrained(
    "microsoft/dit-base-finetuned-rvlcdip",
    device_map="auto",
)
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/dit-example.jpg"
image = Image.open(requests.get(url, stream=True).raw)
inputs = image_processor(image, return_tensors="pt").to("cuda")

with torch.no_grad():
  logits = model(**inputs).logits
predicted_class_id = logits.argmax(dim=-1).item()

class_labels = model.config.id2label
predicted_class_label = class_labels[predicted_class_id]
print(f"The predicted class label is: {predicted_class_label}")
```

File: docs/source/en/model_doc/dit.md [hunk 9]
```diff
@@ -13,80 +13,86 @@ specific language governing permissions and limitations under the License.
...
 
-<Tip>
+## Notes
 
-  As DiT's architecture is equivalent to that of BEiT, one can refer to [BEiT's documentation page](beit) for all tips, code examples and notebooks.
-</Tip>
+- **DiT is trained for document image classification**, not OCR. It won’t extract text or do layout parsing.
+- **Input images should be RGB** and typically resized to 224x224 (or 384x384 for large DiT models).
+- The DiT models are **based on the ViT architecture, with self-supervised pretraining** on large-scale document datasets.
+- As **DiT's architecture is equivalent to that of BEiT**, one can refer to [BEiT's documentation page](beit.md) for all tips, code examples and notebooks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dit.md]:
```suggestion
- The pretrained DiT weights can be loaded in a [BEiT] model with a modeling head to predict visual tokens.

   ```py
   from transformers import BeitForMaskedImageModeling
   
   model = BeitForMaskedImageModeling.from_pretraining("microsoft/dit-base")
   ```
```

Comment (stevhliu) [reply docs/source/en/model_doc/dit.md]: Let's also add a resource section

```md
## Resources

- Refer to this [notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/DiT/Inference_with_DiT_(Document_Image_Transformer)_for_document_image_classification.ipynb) for a document image classification inference example.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for the new model card!

**Comment by Vixel2006** [comment]
Thanks for the guidance [@stevhliu](https://github.com/stevhliu).

**Comment by stevhliu** [comment]
There are still a few unresolved comments and you don't need to make any changes to the `toctree` either

**Comment by Vixel2006** [comment]
[@stevhliu](https://github.com/stevhliu) Thanks again! All tests are passing now. Everything should be ready for merge let me know if anything else comes up.

**Review by stevhliu — APPROVED** [review]
🤗
